MyToolsYourToolsClient/src/app/user-profile/opinions/opinions.component.ts

```diff
@@ -18,12 +19,15 @@ export class OpinionsComponent implements OnInit {
   constructor(private opinionService: OpinionService, private userService: UserService) { }
 
   ngOnInit() {
-  this.opinionService.getOpinions().subscribe(o => this.opinions = o);
-  this.userService.getUsers().subscribe(u => this.users = u);
-  this.opinions = this.opinions.filter( o => o.ratedUserId === this.user.id);
+    this.userService.getUsers().subscribe(u => this.users = u);
+    let allOpinions: Opinion[];
+    this.opinionService.getOpinions()
+      .pipe(
+        tap(allOpinions => this.opinions = allOpinions.filter( o => o.ratedUserId === this.user.id))
+      )
```

Comment:
to: - let allOpinions: Opinion[]; - jest nie potrzebne wiesz, wiem że ja wcześniej też tak miałem ale to przez to ze zapomniałem tego usunąć, 
ta funkcja tap, działa tak samo jak subscribe, czyli zwykła lambda która w tym argumencie ma to co zwraca kontroler z API, to tak na przyszłość
ogólnie fajnie że zrobiłaś to przez pipe'a